middleware.js

```diff
@@ -9,6 +9,34 @@ function asyncMiddleware (fn) {
   };
 }
 
+
+function logErrors (err, req, res, next) {
+  console.log("logErrors");
+  console.error(err.stack);
+  next(err);
+}
+
+
+function clientErrorHandler (err, req, res, next) {
+  console.log("clientErrorHandler");
+  if (req.xhr) {
+    res.status(500).send({ error: err });
```

Comment:
Should.... Should this be `{ error: err }`, or `{error: err.message }`?
How much stack info are we meaning to potentially disclose?